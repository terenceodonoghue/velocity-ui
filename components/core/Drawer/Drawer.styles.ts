import { css, FunctionInterpolation, Theme } from '@emotion/react';

const drawer: FunctionInterpolation<Theme> = ({
  palette,
  transitions,
  zIndex,
}) => css`
  position: fixed;
  top: 80px;
  bottom: 0;
  padding: 20px 0;
  background-color: ${palette.neutral[50]};
  overflow-x: hidden;
  transition: width ${transitions.duration.standard}ms
    ${transitions.easing.easeInOut};
  z-index: ${zIndex.drawer};
`;

export default drawer;
