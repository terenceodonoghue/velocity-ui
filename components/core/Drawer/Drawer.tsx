import React, { FunctionComponent } from 'react';
import drawer from './Drawer.styles';
import type { DrawerProps } from './Drawer.types';

const Drawer: FunctionComponent<DrawerProps> = ({
  anchor = 'left',
  children,
  open = false,
  ...props
}) => (
  <div
    css={[
      drawer,
      anchor === 'left' ? { left: 0 } : { right: 0 },
      {
        width: open ? '215px' : '80px',
      },
    ]}
    data-testid="drawer"
    {...props}
  >
    {children}
  </div>
);

export default Drawer;
