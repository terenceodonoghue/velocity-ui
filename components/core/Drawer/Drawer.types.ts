import { HTMLAttributes } from 'react';

export interface DrawerProps extends HTMLAttributes<HTMLDivElement> {
  anchor?: 'left' | 'right';
  open?: boolean;
}
