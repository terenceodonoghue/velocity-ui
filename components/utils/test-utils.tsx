import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
