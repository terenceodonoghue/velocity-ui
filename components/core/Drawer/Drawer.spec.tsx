import React from 'react';
import { render } from 'test-utils';
import Drawer from './Drawer';

describe('Drawer', () => {
  it('is closed by default', () => {
    const { container, getByTestId } = render(<Drawer />);

    expect(getByTestId('drawer')).toHaveStyle({ width: '80px' });
    expect(container).toMatchSnapshot();
  });

  it('expands when opened', () => {
    const { container, getByTestId } = render(<Drawer open />);

    expect(getByTestId('drawer')).toHaveStyle({ width: '215px' });
    expect(container).toMatchSnapshot();
  });

  it('anchors to the left', () => {
    const { container, getByTestId } = render(<Drawer anchor="left" />);

    expect(getByTestId('drawer')).toHaveStyle({ left: 0 });
    expect(container).toMatchSnapshot();
  });

  it('anchors to the right', () => {
    const { container, getByTestId } = render(<Drawer anchor="right" />);

    expect(getByTestId('drawer')).toHaveStyle({ right: 0 });
    expect(container).toMatchSnapshot();
  });
});
