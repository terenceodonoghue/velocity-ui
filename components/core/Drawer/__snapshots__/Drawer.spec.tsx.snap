// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer anchors to the left 1`] = `
.emotion-0 {
  position: fixed;
  top: 80px;
  bottom: 0;
  padding: 20px 0;
  background-color: #fff;
  overflow-x: hidden;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1);
  z-index: 1200;
  left: 0;
  width: 80px;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer"
  />
</div>
`;

exports[`Drawer anchors to the right 1`] = `
.emotion-0 {
  position: fixed;
  top: 80px;
  bottom: 0;
  padding: 20px 0;
  background-color: #fff;
  overflow-x: hidden;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1);
  z-index: 1200;
  right: 0;
  width: 80px;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer"
  />
</div>
`;

exports[`Drawer expands when opened 1`] = `
.emotion-0 {
  position: fixed;
  top: 80px;
  bottom: 0;
  padding: 20px 0;
  background-color: #fff;
  overflow-x: hidden;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1);
  z-index: 1200;
  left: 0;
  width: 215px;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer"
  />
</div>
`;

exports[`Drawer is closed by default 1`] = `
.emotion-0 {
  position: fixed;
  top: 80px;
  bottom: 0;
  padding: 20px 0;
  background-color: #fff;
  overflow-x: hidden;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1);
  z-index: 1200;
  left: 0;
  width: 80px;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer"
  />
</div>
`;
